pkg/api/v1/types/types.go

```diff
@@ -51,20 +53,67 @@ type ConditionUpdate struct {
 	ResourceVersion int64                 `json:"resourceVersion"`
 }
 
+func (c *ConditionUpdate) Validate() error {
+	if c.ResourceVersion == 0 {
+		return errors.Wrap(errUpdatePayload, "ResourceVersion not set")
+	}
+
+	if c.State == "" || c.Status == nil {
+		return errors.Wrap(errUpdatePayload, "state and status attributes are expected")
+	}
+
+	return nil
+}
+
+// ConditionUpdateEvent is the payload received for a condition update over the event stream.
+type ConditionUpdateEvent struct {
+	ConditionUpdate
+	Kind ptypes.ConditionKind `json:"kind"`
+}
+
+// Validate checks for required attributes.
+//
+// Note:
+// The ResourceVersion attribute is not validated for updates through events,
+// this is because the controllers do not perform requests for the existing condition
+// since implementing a Request-Reply pattern on the NATS Jetstream is tideous and not recommended.
+//
+// The NATS Jetstream guarantees ordered delivery, as long as there is a single consumer of the event,
+// for now we're deploying a single orchestrator instance in each facility and so this check is not required.
+// In the case that we require multiple orchestrators in a facility the stream would need to be partitioned
+// to ensure ordered delivery.
+//
+// ref:
+// https://github.com/nats-io/nats.py/discussions/221
+// https://github.com/nats-io/nats.go/discussions/970#discussioncomment-2690789
+//
+// TODO: move this note into the messaging architecture doc.
+func (c *ConditionUpdateEvent) Validate() error {
+	if c.Kind == "" {
+		return errors.Wrap(errUpdatePayload, "Kind attribute expected")
+	}
+
+	if c.State == "" || c.Status == nil {
+		return errors.Wrap(errUpdatePayload, "state and status attributes are expected")
+	}
+
+	return nil
+}
+
 // MergeExisting when given an existing condition, validates the update based on existing values
 // and returns a condition that can be passed to the repository for update.
 //
 // The resourceVersion is not updated here and is left for the repository Store to update.
 //
 // This method makes sure that update does not overwrite existing data inadvertently.
```

Comment:
why would we want to disregard the resource versions on a merge?

Reply:
I see what you're getting at, but I wonder if maybe we need a better mechanism for managing or protecting against concurrent access.